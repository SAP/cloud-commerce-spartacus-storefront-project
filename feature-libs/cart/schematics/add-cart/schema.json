{
  "$schema": "http://json-schema.org/schema",
  "id": "CartSchematics",
  "title": "Cart Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Saved cart feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": ["Saved-cart"],
        "type": "string"
      },
      "default": ["Saved-cart"],
      "x-prompt": "Which features would you like to setup?"
    }
  },
  "required": []
}
