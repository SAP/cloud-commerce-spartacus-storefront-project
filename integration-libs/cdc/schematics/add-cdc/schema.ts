import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  baseSite?: string;
  jsSDKUrl?: string;
  sessionExpiration?: number;
}
