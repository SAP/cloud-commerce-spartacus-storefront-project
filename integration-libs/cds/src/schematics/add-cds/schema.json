{
  "$schema": "http://json-schema.org/schema",
  "id": "CDSSchematics",
  "title": "CDS Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "tenant": {
      "type": "string",
      "description": "The tenant name.",
      "default": "my-tenant",
      "x-prompt": "What is your tenant name?"
    },
    "baseUrl": {
      "type": "string",
      "description": "The base URL.",
      "default": "https://my-tenant.api.us.context.cloud.sap",
      "x-prompt": "What is your base URL?"
    },
    "profileTagLoadUrl": {
      "type": "string",
      "description": "Where to load ProfileTag from. (e.g. https://tag.static.us.context.cloud.sap/js/profile-tag.js)",
      "x-prompt": "Where to load ProfileTag from? (optional)"
    },
    "profileTagConfigUrl": {
      "type": "string",
      "description": " Configuration URL for ProfileTag. (e.g. https://tag.static.stage.context.cloud.sap/config/my-config123)",
      "x-prompt": "What is the ProfileTag configuration URL? (optional)"
    }
  },
  "required": ["tenant", "baseUrl"]
}
