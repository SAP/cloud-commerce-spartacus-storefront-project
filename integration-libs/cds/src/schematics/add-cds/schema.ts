import { LibraryOptions } from '@spartacus/schematics';

export interface Schema extends LibraryOptions {
  tenant: string;
  baseUrl: string;
  strategyId: string;
  profileTagLoadUrl?: string;
  profileTagConfigUrl?: string;
}
