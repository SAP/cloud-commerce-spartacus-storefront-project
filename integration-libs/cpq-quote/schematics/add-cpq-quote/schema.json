{
  "$schema": "http://json-schema.org/schema",
  "$id": "CpqQuoteSchematics",
  "title": "cpq-quote Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display Cpq Discount Percentage",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the cpq-quote.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["cpq-quote"]
    }
  },
  "required": []
}
