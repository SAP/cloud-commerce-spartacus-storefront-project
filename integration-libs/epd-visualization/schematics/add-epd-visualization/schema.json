{
  "$schema": "http://json-schema.org/schema",
  "$id": "EpdVisualizationSchematics",
  "title": "EpdVisualization Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["EpdVisualization"]
    },
    "baseUrl": {
      "type": "string",
      "description": "Enter the base URL (origin) of your EPD Fiori Launchpad",
      "default": "https://epd-dev-eu20-consumer.epddev.cfapps.eu20.hana.ondemand.com",
      "x-prompt": "[EPD Visualization] What is the base URL (origin) of your EPD Fiori Launchpad?"
    }
  },
  "required": []
}
