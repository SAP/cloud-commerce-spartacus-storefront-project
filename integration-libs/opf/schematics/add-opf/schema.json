{
  "$schema": "http://json-schema.org/schema",
  "$id": "OpfSchematics",
  "title": "Opf Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the opf features.",
      "default": false
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["OPF"]
    },
    "baseUrl": {
      "type": "string",
      "description": "The base url of Cloud Commerce Adapter for Open Payment Framework integration",
      "x-prompt": "[OPF] What is the base URL (origin) of your OPF Cloud Commerce Adapter?"
    },
    "commerceCloudPublicKey": {
      "type": "string",
      "description": "Commerce Clould public key required for authentication between OPF Cloud Commerce Adapter and Spartacus"
    }
  },
  "required": []
}
