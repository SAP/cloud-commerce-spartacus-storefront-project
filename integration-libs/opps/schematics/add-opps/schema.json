{
  "$schema": "http://json-schema.org/schema",
  "$id": "OppsSchematics",
  "title": "OPPS Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the OPPS features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Opps-Coupon-Codes", "Opps-Login-Required"],
      "x-prompt": {
        "message": "Which OPPS features would you like to set up?",
        "type": "list",
        "items": [
          {
            "value": "Opps-Coupon-Codes",
            "label": "OPPS Coupon Codes"
          },
          {
            "value": "Opps-Login-Required",
            "label": "OPPS - Recommend Users To Login To Get More Promotions"
          }
        ]
      }
    }
  },
  "required": []
}
