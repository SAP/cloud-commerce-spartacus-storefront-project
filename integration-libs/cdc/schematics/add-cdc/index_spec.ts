/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  CLI_CDC_FEATURE,
  LibraryOptions as SpartacusCdcOptions,
  SpartacusOptions,
} from '@spartacus/schematics';
import * as path from 'path';
import featureLibPackageJson from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');
const cdcModulePath = 'src/app/spartacus/features/cdc/cdc-feature.module.ts';

describe('Spartacus CDC schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    configuration: 'b2c',
    lazy: true,
    features: [],
  };

  const defaultFeatureOptions: SpartacusCdcOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [CLI_CDC_FEATURE],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      '@spartacus/schematics',
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@spartacus/schematics',
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('When no features are provided', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync(
          'ng-add',
          { ...defaultFeatureOptions, features: [] },
          appTree
        )
        .toPromise();
    });

    it('should not create the feature module', () => {
      const featureModule = appTree.readContent(cdcModulePath);
      expect(featureModule).toBeFalsy();
    });
    it('should not add the feature to the feature module', () => {
      const spartacusFeaturesModule = appTree.readContent(
        'src/app/spartacus/spartacus-features.module.ts'
      );
      expect(spartacusFeaturesModule).toMatchSnapshot();
    });
  });

  describe('CDC feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should install @spartacus/asm and @spartacus/user', async () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        expect(packageJson.dependencies['@spartacus/asm']).toEqual(
          `^${featureLibPackageJson.peerDependencies['@spartacus/asm']}`
        );
        expect(packageJson.dependencies['@spartacus/user']).toEqual(
          `^${featureLibPackageJson.peerDependencies['@spartacus/user']}`
        );
      });

      it('should import feature module in SpartacusFeaturesModule', () => {
        const spartacusFeaturesModulePath = appTree.readContent(
          'src/app/spartacus/spartacus-features.module.ts'
        );
        expect(spartacusFeaturesModulePath).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import correct modules (without lazy loaded syntax)', async () => {
        const cdcModule = appTree.readContent(cdcModulePath);
        expect(cdcModule).toMatchSnapshot();
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultFeatureOptions, appTree)
          .toPromise();
      });

      it('should import correct modules (with lazy loaded syntax)', async () => {
        const cdcModule = appTree.readContent(cdcModulePath);
        expect(cdcModule).toMatchSnapshot();
      });
    });
  });
});
