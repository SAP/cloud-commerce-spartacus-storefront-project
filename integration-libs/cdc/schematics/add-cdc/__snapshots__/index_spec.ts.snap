// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus CDC schematics: ng-add CDC feature eager loading should import correct modules (without lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { CdcModule } from \\"@spartacus/cdc\\";
import { CdcConfig, CdcRootModule } from \\"@spartacus/cdc/root\\";
import { provideConfig } from \\"@spartacus/core\\";

@NgModule({
  declarations: [],
  imports: [
    CdcRootModule,
    CdcModule
  ],
  providers: [provideConfig(<CdcConfig>{
    cdc: [
      {
        baseSite: 'electronics-spa',
        javascriptUrl: '<url-to-cdc-script>',
        sessionExpiration: 3600
      },
    ],
  })]
})
export class CdcFeatureModule { }
"
`;

exports[`Spartacus CDC schematics: ng-add CDC feature general setup should import feature module in SpartacusFeaturesModule 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CartModule, CartOccModule, CheckoutModule, CheckoutOccModule, CostCenterOccModule, ProductModule, ProductOccModule, UserOccTransitionalModule, UserTransitionalModule } from \\"@spartacus/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CartComponentModule, CartPageEventModule, CategoryNavigationModule, CheckoutComponentModule, CheckoutLoginModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, OrderCancellationModule, OrderConfirmationModule, OrderDetailsModule, OrderHistoryModule, OrderReturnModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ReplenishmentOrderConfirmationModule, ReplenishmentOrderDetailsModule, ReplenishmentOrderHistoryModule, ReturnRequestDetailModule, ReturnRequestListModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule, WishListModule } from \\"@spartacus/storefront\\";
import { CdcFeatureModule } from './features/cdc/cdc-feature.module';

@NgModule({
  declarations: [],
  imports: [
    // Auth Core
    AuthModule.forRoot(),
    LogoutModule,
    LoginRouteModule,
    // Basic Cms Components
    HamburgerMenuModule,
    SiteContextSelectorModule,
    LinkModule,
    BannerModule,
    CmsParagraphModule,
    TabParagraphContainerModule,
    BannerCarouselModule,
    CategoryNavigationModule,
    NavigationModule,
    FooterNavigationModule,
    BreadcrumbModule,
    // User Core,
    UserTransitionalModule,
    UserOccTransitionalModule,
    // User UI,
    AddressBookModule,
    PaymentMethodsModule,
    NotificationPreferenceModule,
    MyInterestsModule,
    StockNotificationModule,
    ConsentManagementModule,
    MyCouponsModule,
    // Anonymous Consents Core,
    AnonymousConsentsModule.forRoot(),
    // Anonymous Consents UI,
    AnonymousConsentsDialogModule,
    AnonymousConsentManagementBannerModule,
    // Product Core,
    ProductModule.forRoot(),
    ProductOccModule,
    // Product UI,
    ProductDetailsPageModule,
    ProductListingPageModule,
    ProductListModule,
    SearchBoxModule,
    ProductFacetNavigationModule,
    ProductTabsModule,
    ProductCarouselModule,
    ProductReferencesModule,
    ProductImagesModule,
    ProductSummaryModule,
    ProductIntroModule,
    // Cart Core,
    CartModule.forRoot(),
    CartOccModule,
    // Cart UI,
    CartComponentModule,
    WishListModule,
    // Checkout Core,
    CheckoutModule.forRoot(),
    CheckoutOccModule,
    CostCenterOccModule,
    // Checkout UI,
    CheckoutLoginModule,
    CheckoutComponentModule,
    OrderConfirmationModule,
    // Order,
    OrderHistoryModule,
    OrderDetailsModule,
    OrderCancellationModule,
    OrderReturnModule,
    ReturnRequestListModule,
    ReturnRequestDetailModule,
    ReplenishmentOrderHistoryModule,
    ReplenishmentOrderDetailsModule,
    ReplenishmentOrderConfirmationModule,
    // Page Events,
    NavigationEventModule,
    HomePageEventModule,
    CartPageEventModule,
    ProductPageEventModule,
    CdcFeatureModule,
  ]
})
export class SpartacusFeaturesModule { }
"
`;

exports[`Spartacus CDC schematics: ng-add CDC feature lazy loading should import correct modules (with lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { CdcConfig, CdcRootModule, CDC_FEATURE } from \\"@spartacus/cdc/root\\";
import { CmsConfig, provideConfig } from \\"@spartacus/core\\";

@NgModule({
  declarations: [],
  imports: [
    CdcRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [CDC_FEATURE]: {
        module: () =>
          import('@spartacus/cdc').then((m) => m.CdcModule),
      },
    }
  }),
  provideConfig(<CdcConfig>{
    cdc: [
      {
        baseSite: 'electronics-spa',
        javascriptUrl: '<url-to-cdc-script>',
        sessionExpiration: 3600
      },
    ],
  })
  ]
})
export class CdcFeatureModule { }
"
`;
